pkg/cluster/overridemtu.go

```diff
@@ -0,0 +1,94 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/machines/machineconfig"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/cluster/graph"
+)
+
+const (
+	IgnFilePath = "/etc/NetworkManager/dispatcher.d/30-eth0-mtu-3900"
+	IgnFileUser = "root"
+	IgnFileMode = 0555
+	IgnFileData = `#!/bin/bash
+
+if [ "$1" == "eth0" ] && [ "$2" == "up" ]; then
+    ip link set $1 mtu 3900
+fi`
+)
+
+func (m *manager) overrideEthernetMTU(g graph.Graph) error {
+	ignFile := ignition.FileFromString(IgnFilePath, IgnFileUser, IgnFileMode, IgnFileData)
+
+	// Extend Ignition config for master nodes
+
+	master := g.Get(&machine.Master{}).(*machine.Master)
+	master.Config.Storage.Files = append(master.Config.Storage.Files, ignFile)
```

Comment:
Not familiar with what this is doing but my read of it is some special configuration is landing directly on masters, bypassing kubernetes.  Is that true? Is it necessary?  Could MTU be managed for masters in the same way it's managed for workers?

Reply:
> Could MTU be managed for masters in the same way it's managed for workers?

Possibly.  To some extent I went with the easiest Installer API that worked, and overriding MTU on masters was pretty straight-forward.  Overriding MTU on worker nodes required a couple more levels of indirection on the bootstrap node, as you can see in the code.

I believe this works for masters because of [this code in the ARO-RP](https://github.com/Azure/ARO-RP/blob/0b39eb5de7f5b3a9e895845b7485f7ebe74e09bf/pkg/cluster/deployresources_resources.go#L187), where the master ignition config is passed directly as the "CustomData" part of the Azure VM resource definition.

I can try the MachineConfig approach for masters if consistency with workers is important.

Reply:
I think it's important when we look at day-2 operations.  If the way you have it allows for that then I'm less concerned but there are inconsistencies.  And reading this I have no idea what day-2 for master management would be if we needed it.  Am I missing something?

Reply:
I think I'm missing what day-2 operations you had in mind for masters.

Nevertheless, I'll try an alternate method for masters tomorrow.

Reply:
Well, 2 kind: create or update.
If we create a new master (i.e. one got deleted) will this work easily?
If we update MTU for some reason will we be able to that on the masters easily?

Reply:
> If we create a new master (i.e. one got deleted) will this work easily?
I believe so, because this code embeds the ignition config in the Azure VM resource definition for master nodes (in the CustomData field).  But I'm testing a different approach for master nodes as you requested, so this point may be moot.

> If we update MTU for some reason will we be able to that on the masters easily?
Until *cluster-network-operator* supports changing the MTU in SDN/OVN, we should forget about changing MTU on day 2.  I went down that rabbit hole already.